### PR TITLE
EHUB-383 - test fix

### PR DESCRIPTION
**Pull request by MEM2677**
Due to the non-working pipeline we ended up with some tests failing on the develop branch, we have to fix them

### CODE REVIEW

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java [hunk 1]
```diff
@@ -17,25 +17,26 @@
...
         HttpStatus status;
         if ((exception instanceof BadRequestException) ||
-            (exception instanceof IllegalArgumentException)) {
+                (exception instanceof IllegalArgumentException)) {
             status = HttpStatus.BAD_REQUEST;
         } else if (exception instanceof AccessDeniedException) {
             status = HttpStatus.FORBIDDEN;
-        } else if (exception instanceof NotFoundException) {
+        } else if (exception instanceof NotFoundException
+                || exception instanceof ResponseStatusException) {
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java]: This is not always correct: the ResponseStatusException can contain any status code. In my opinion you have 2 ways to handle the problem:
- separate these 2 cases (NotFoundException and ResponseStatusException) and in the second one you can simply take the status code from the exception
- update the code that throws the ResponseStatusException to throw a NotFoundException

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/BundleGroupFlowIT.java [hunk 2]
```diff
@@ -235,12 +235,14 @@ void shouldUpdateBundleGroup() throws Exception {
...
-        Organisation organisationSaved = organisationRepository.save(new Organisation().setName(ORG_NAME).setDescription(ORG_DESCRIPTION));
+        Organisation organisationSaved = organisationRepository.save(new Organisation()
+                .setName(ORG_NAME)
+                .setDescription(ORG_DESCRIPTION));
         catalogRepository.save(new Catalog().setName(CAT_NAME).setOrganisation(organisationSaved));
         BundleGroup stubBundleGroup = getStubBundleGroup(true, organisationSaved, Optional.empty());
 
         BundleGroupDto bundleGroupNoId = bundleGroupMapper.toDto(stubBundleGroup);
-
+        bundleGroupNoId.setBundleGroupId(null);
```
Comment (firegloves) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/BundleGroupFlowIT.java:245]: Is this code required?

**Review by firegloves — CHANGES_REQUESTED** [review]

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/BundleGroupFlowIT.java (hunk 2 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/BundleGroupFlowIT.java:245]: Yes, because the intention of the developer was to provide an object of type BundleGroup with no ID. The stubbed objcect has a default ID (BUNDLE_GROUP_ID), so to respect the original will of the developer I forced it to null.

File: application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java (hunk 1 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/handlers/RestExceptionHandler.java]: Option #1 seems the best way to go.

**Review by firegloves — APPROVED** [review]
LGTM

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java [hunk 3]
```diff
@@ -280,7 +283,7 @@ void shouldReturnForbiddenWhenDeleteCatalogWithNoAdminRole() throws Exception {
     void shouldReturnUnauthorizedWhenDeleteCatalogWithNoRole() throws Exception {
         mockMvc.perform(delete("/api/catalog/1")
                         .accept(MediaType.APPLICATION_JSON_VALUE))
-                .andExpect(status().isUnauthorized())
+                .andExpect(status().isForbidden())
```
Comment (alepintus) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java:286]: Please rename the method name with something like shouldReturnForbiddenWhenDeleteCatalogWithNoRole

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java [hunk 4]
```diff
@@ -236,7 +239,7 @@ void shouldReturnForbiddenWhenCreateCatalogWithNoAdminRole() throws Exception {
     void shouldReturnUnauthorizedWhenCreateCatalogWithNoRole() throws Exception {
```
Comment (alepintus) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java]: Please rename the method name with something like shouldReturnForbiddenWhenCreateCatalogWithNoRole

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java [hunk 5]
```diff
@@ -194,7 +197,7 @@ void shouldReturnForbiddenWhenGettingWithNonAdminRole() throws Exception {
     void shouldReturnUnauthorizedWhenGettingWithNoRole() throws Exception {
```
Comment (alepintus) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java]: Please rename the method name with something like shouldReturnForbiddenWhenGettingWithNoRole

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java [hunk 6]
```diff
@@ -155,7 +158,7 @@ void shouldGetEmptyListOfCatalogsWhenDbIsEmpty() throws Exception {
     void shouldReturnUnauthorizedForNoRole() throws Exception {
         mockMvc.perform(get("/api/catalog/")
                         .accept(MediaType.APPLICATION_JSON_VALUE))
-                .andExpect(status().isUnauthorized());
+                .andExpect(status().isForbidden());
```
Comment (alepintus) [inline application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java:161]: Please rename the method name with something like shouldReturnForbiddenForNoRole

**Review by alepintus — CHANGES_REQUESTED** [review]

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java (hunk 3 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java:286]: Done!

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java (hunk 4 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java]: Done!

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java (hunk 5 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java]: Done!

File: application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java (hunk 6 above)
Comment (MEM2677) [reply application/microservices/entando-hub-catalog-ms/src/test/java/com/entando/hub/catalog/integration/CatalogFlowIT.java:161]: Done!

**Review by alepintus — APPROVED** [review]

File: .github/workflows/pr.yml [hunk 7]
```diff
@@ -18,13 +18,16 @@ jobs:
         fetch-depth: 0
     - name: SonarQube Scan
       run: |
+        #export JAVA_HOME=/usr/lib/jvm/default-jvm
+        export LD_LIBRARY_PATH=/usr/lib/jvm/java-11-openjdk/lib/server
+        #export PATH=$PATH:$JAVA_HOME/bin
```
Comment (firegloves) [inline .github/workflows/pr.yml]: please remove these lines

File: .github/workflows/pr.yml [hunk 8]
```diff
@@ -18,13 +18,16 @@ jobs:
...
         versionSuffix=$(echo "${{ github.event.pull_request.title }}" | awk '{print $1}')
         bash ../.github/functions/utils.sh setBundleVersion -$versionSuffix
         bash ../.github/functions/utils.sh setComponentVersions -$versionSuffix
         ent bundle build --all --stdout --fail-fast
         cd microservices/entando-hub-catalog-ms && mvn sonar:sonar --no-transfer-progress && cd ../..
-        cd microfrontends/eh-login && ent npm run sonar && cd ../..
-        cd microfrontends/eh-app && ent npm run sonar && cd ../..
+        cd microfrontends/eh-login && ent npm install && ent npm run sonar && cd ../..
+        cd microfrontends/eh-login && ent npm install && ent npm run sonar && cd ../..
```
Comment (firegloves) [inline .github/workflows/pr.yml]: revert these changes please

**Review by firegloves — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yml (hunk 7 above)
Comment (MEM2677) [reply .github/workflows/pr.yml]: Done

File: .github/workflows/pr.yml (hunk 8 above)
Comment (MEM2677) [reply .github/workflows/pr.yml]: Done

**Review by firegloves — APPROVED** [review]
LGTM
